ext/storage_sources/gcp_store/gcp_storage_source.cpp

```diff
@@ -192,9 +196,26 @@ gcp_add_reference(WT_STORAGE_SOURCE *storage_source)
 static int
 gcp_file_close(WT_FILE_HANDLE *file_handle, WT_SESSION *session)
 {
-    WT_UNUSED(file_handle);
     WT_UNUSED(session);
 
+    gcp_file_handle *gcp_fh = reinterpret_cast<gcp_file_handle *>(file_handle);
+
+    // If there are other active instances of the file being open, do not close file handle.
+    if (--gcp_fh->reference_count != 0) {
+        return 0;
+    }
+
+    // No more active instances of open file, close the file handle.
+    gcp_file_system *fs = reinterpret_cast<gcp_file_handle *>(gcp_fh)->file_system;
```

Comment:
Is there a reason why we're doing this `reinterpret_cast` thing here? Will `gcp_fh->file_system` work here?

Reply:
I agree `gcp_fh->file_system` is more suitable here, fixed in: [9faf10f](https://github.com/wiredtiger/wiredtiger/pull/8777/commits/9faf10fe9b57e6c7c29f5909f2cb8c35c5a3ec8e)